### ISSUE TITLE
Broken demo page - Input section

### ISSUE THREAD

**Issue by Shyiy** [angular/material#6822]
Version: Head/Master

By following a very simple series of steps on the demos webpage it is possible to trigger an error, which then causes all navigation to stop working until the page is reloaded:
```
TypeError: Cannot read property 'runner' of undefined
    at angular-animate.js:2508
    at n (angular.js:340)
    at y (angular-animate.js:2503)
    at l (angular-animate.js:2313)
    at Object.push (angular-animate.js:2194)
    at Object.leave (angular.js:5302)
    at z (angular-route.js:913)
    at angular-route.js:942
    at angular.js:7610
    at angular.js:7749
```

Steps:
- Navigate to https://material.angularjs.org/HEAD/demo/input
- Focus one of the inputs in the "Errors Advanced" or "Icons" containers (can also click other things afterwards)
- Try to navigate to another demo page

**Comment by ThomasBurleson** [comment]
May be related to `$destroy` handlers.

**Comment by DevVersion** [comment]
Fixed in #6880
